### DISCUSSION TITLE
How can I make package to reference types from generated prisma-client from custom path

### DISCUSSION THREAD

**Discussion by dovigod** [prisma/prisma#25565] [Q&A]
### Question

Hi, I don’t think this question is directly related to Prisma, but I'm posting this discussion in the hope that someone might have an answer.

I'm currently developing on monorepo, implementing internal module which has additional features base on [nestjs-prisma](https://www.npmjs.com/package/nestjs-prisma)

project structure is as following

````
     - apps
    |    |_serviceA
    |        |_node_modules
    |             |_myPrismaModule (install built version of internal)
    |        |_src
    |           |_generated
    |               |_client
    |                   |_default.js (where my required type lies in)
    |
    |- internal
    |     |_myPrismaModule
````

Since I'm using [Prisma](https://www.prisma.io/?via=start&gad_source=1), PrismaClient & Prisma type will defer across all service , cuz its type is defined by how developer designs the prisma schema.

The problem I'm facing is that I can't reference the type properly after interna module build.. (actually I can't guarantee whether this problem is reference problem or not..)

My implementation is the following code, previously it worked well but after changes it started not to reference well.

````
//previous
import {Prisma, PrismaClient} from "@prisma/client"

@Injectable()
export class PrismaService
  extends PrismaClient<Prisma.PrismaClientOptions, 'query' | 'info' | 'warn' | 'error'>
  implements OnModuleInit{...}


------

// current


async function getPrismaClientType() {
  const {Prisma, PrismaClient} = await import(
    "~~absolutePath~~/src/apps/serviceA/src/generated/client/default.js'"
  );
  return PrismaClient as PrismaClient<
   Prisma.PrismaClientOptions,
    'query' | 'info' | 'warn' | 'error'
  >;
}
@Injectable()
export class PrismaService
  extends (await getPrismaClientType())
  implements OnModuleInit{...}

````

It throwed Cannot find namespace 'Prisma' but I just to proceed, I commented @ts-ignore in this case. (suspicious, but if so, how can I handle this?)

And In service workspace, whenever I try to access type like below, compile infer type as "any".. any advise in this??

```
 prismaService.banner.findMany // banner is any
````

Also wrote [same question](https://stackoverflow.com/questions/79147164/how-to-reference-an-modules-type-from-application-which-uses-this-module) in stack-overflow, but none answered yet.

### Expected behavior (optional)

after importing internal prisma module (which provides reference to generated `prisma-client`), types corresponding to services's schema should be supported

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by nurul3101** [comment]
Hello @dovigod 👋 

To make sure that I understand the issue correctly - Your setup worked as expected when PrismaClient and Prisma types were generated in the default location (inside node_modules). However, once you configured a custom path by setting output parameter in `schema.prisma` file as mentioned in the snippet below, you started getting the `Cannot find namespace 'Prisma' ` error, correct?

```prisma
generator client {
  provider = "prisma-client-js"
  output   = "./generated/client"
}
```

**Reply by dovigod** [reply]
Yap exactly :)

**Reply by nurul3101** [reply]
Thanks for confirming. Did you see this related GitHub Issue?
- https://github.com/prisma/prisma/issues/21066

 I think you are running into this. Can you temporarily try setting custom path in node_modules and check if that works?

**Reply by dovigod** [reply]
@nurul3101 Thanks for the suggestion :) I think this is it. It worked when I set it inside `node_modules`. But just to confirm, is there any way to set a custom path in the root directory in a monorepo environment?

**Reply by dovigod** [reply]
And could you also let me know why [this message with your suggestion](https://github.com/prisma/prisma/issues/21066#issuecomment-1718141778) is the case? As far as I know, it should work properly since I've set the prismaClient import as an absolute path.

**Reply by nurul3101** [reply]
Absolute path should ideally work, but I assume this is a bug on our end or something going wrong with NestJS which is why the linked GitHub Issue is still open.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
